test/test_socket_options.cpp

```diff
@@ -850,6 +850,15 @@ TEST_F(TestSocketOptions, StreamIDWrongLen)
     EXPECT_EQ(srt_getlasterror(NULL), SRT_EINVPARAM);
 }
 
+//Check if setting -1 as optlen returns an error 
+TEST_F(TestSocketOptions, StringOptLenInvalid)
+{
+
+    string stream_id = "test123";
```

Comment:
Although the scope is short, still the `stream_id` variably can be `const`.
```suggestion
    const string stream_id = "test123";
```